zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -165,19 +168,34 @@ impl UpdateWith<PreparedBlock> for Chain {
             .zip(transaction_hashes.iter().cloned())
             .enumerate()
         {
-            use transaction::Transaction::*;
-            let (inputs, joinsplit_data, sapling_shielded_data) = match transaction.deref() {
-                V4 {
-                    inputs,
-                    joinsplit_data,
-                    sapling_shielded_data,
-                    ..
-                } => (inputs, joinsplit_data, sapling_shielded_data),
-                V5 { .. } => unimplemented!("v5 transaction format as specified in ZIP-225"),
-                V1 { .. } | V2 { .. } | V3 { .. } => unreachable!(
-                    "older transaction versions only exist in finalized blocks pre sapling",
-                ),
-            };
+            let (inputs, joinsplit_data, sapling_shielded_data, orchard_shielded_data) =
+                match transaction.deref() {
+                    V4 {
+                        inputs,
+                        joinsplit_data,
+                        sapling_shielded_data,
+                        ..
+                    } => (
+                        inputs,
+                        joinsplit_data,
+                        Either::Left(sapling_shielded_data),
```

Comment:
I will be happy using something else than `Either` here but i was not able to find a better alternative.

Reply:
I suggest splitting the sapling_shielded_data field into two fields:
* sapling_shielded_data_per_spend_anchor
* sapling_shielded_data_shared_anchor

Then you can pass None for per spend in V5, and shared in V4.

Rust treats these generic types with different type parameters as two different types. So it will be easiest for us to do the same thing here.

(We could erase the types using `&dyn UpdateWith<Chain>`, but that risks getting future fields in the wrong places.)